docs(picker): updated docs with allowed menu content (#5446)
adobe/spectrum-web-components: packages/picker/README.md

### Anatomy

> A picker includes a label and a menu..

Reviewer: Let's change `..` to `.` ✨

> A picker includes a label and a menu..

Reviewer 2: ```suggestion
A picker includes a label and a menu.
```

</sp-tab-panel>
</sp-tabs>

#### Menu

> The picker menu is a menu element that is used to display the options for the picker. A picker menu can include menu items, menu dividers, and menu groups. A picker menu should never contain submenus, as doing so would render it inaccessible.

Author: Notes that submenus would make a picker inaccessible.

> The picker menu is a menu element that is used to display the options for the picker. A picker menu can include menu items, menu dividers, and menu groups. A picker menu should never contain submenus, as doing so would render it in accessible.

Author: Notes that submenus would make a picker inaccessible.

> If you require a submenu, use and [action menu](./action-menu) instead of a picker.

Reviewer 2: 🎶  I loveeeeee this so much

```html demo

> <sp-picker>

Author: Example with menu groups since picker can accept them.

</sp-tab-panel>
</sp-tabs>

#### Do not use submenus

> A picker menu should never contain submenus, as doing so would render it in accessible. A picker's menu role is a listbox and it's menu items are listbox options, which are not allowed to have submenus.

Author: A more detailed explanation and reminder not to use submenus in a picker.

Reviewer 2: Reinforcement 🤌 TY

> A picker menu should never contain submenus, as doing so would render it in accessible. A picker's menu role is a listbox and it's menu items are listbox options, which are not allowed to have submenus.

Reviewer: Should this be `inaccessible`? ✨

> A picker menu should never contain submenus, as doing so would render it in accessible. A picker's menu role is a listbox and it's menu items are listbox options, which are not allowed to have submenus.

Reviewer 2: ```suggestion
A picker menu should never contain submenus, as doing so would render it inaccessible. A picker's menu role is a listbox, and its menu items are listbox options, which are not allowed to have submenus.
```
